src/main/java/org/opensearch/geospatial/action/upload/UploadMetric.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.geospatial.action.upload;
+
+import java.io.IOException;
+
+import org.opensearch.common.xcontent.ToXContentFragment;
+import org.opensearch.common.xcontent.XContentBuilder;
+
+/**
+ * UploadMetric stores metric for an upload API
+ */
+public final class UploadMetric implements ToXContentFragment {
+
+    public enum FIELDS {
+        id,
+        count,
+        success,
+        failed,
+        duration
+    }
+
+    private final long duration;
+    private final long failedCount;
+    private final String metricID;
+    private final long successCount;
+    private final long uploadCount;
+
+    private UploadMetric(UploadMetricBuilder builder) {
+        this.metricID = builder.metricID;
+        this.uploadCount = builder.uploadCount;
+        this.successCount = builder.successCount;
+        this.failedCount = builder.failedCount;
+        this.duration = builder.duration;
+    }
+
+    /**
+     * @return Total number of documents that are failed to upload
+     */
+    public long getFailedCount() {
+        return failedCount;
+    }
+
+    /**
+     * @return Total number of documents that are successfully uploaded
+     */
+    public long getSuccessCount() {
+        return successCount;
+    }
+
+    /**
+     * @return Total number of documents to be uploaded
+     */
+    public long getUploadCount() {
+        return uploadCount;
+    }
+
+    /**
+     * @return Metric's Identifier
+     */
+    public String getMetricID() {
+        return metricID;
+    }
+
+    /**
+     * @return Total time spent in milliseconds to upload, this includes both
+     * succeeded and failed time
+     */
+    public long getDuration() {
+        return duration;
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.field(FIELDS.id.name(), metricID);
+        builder.field(FIELDS.count.name(), uploadCount);
+        builder.field(FIELDS.success.name(), successCount);
+        builder.field(FIELDS.failed.name(), failedCount);
+        builder.field(FIELDS.duration.name(), duration);
+        return builder;
+    }
+
+    /**
+     * Builder to create {@link UploadMetric}
+     */
+    public static final class UploadMetricBuilder {
+
+        private long duration;
+        private long failedCount;
+        private String metricID;
+        private long successCount;
+        private long uploadCount;
```

Comment:
Do these need to be initialized? Seems like metricID at the least should be a required parameter. What happens if this isnt set?

Reply:
Now metric id is required now. Default for all others are 0